SportHub.Business/Implementations/JwtService.cs

```diff
@@ -0,0 +1,132 @@
+                SecurityAlgorithms.HmacSha256Signature)
+        };
+        
+        var accessToken = tokenHandler.CreateToken(tokenDescriptor);
+        
+        tokenDescriptor.Expires = DateTime.UtcNow.AddMinutes(2);
+        var refreshToken  = tokenHandler.CreateToken(tokenDescriptor);
+        _WriteTokenInDb(tokenHandler.WriteToken(refreshToken), user.Email);
+        
+        
+        JwtResponse response= new JwtResponse();
+        response.AccessToken = tokenHandler.WriteToken(accessToken);
+        response.RefreshToken = tokenHandler.WriteToken(refreshToken);
+        response.User = new UserDto
+        {
+            Email = user.Email,
+            FirstName = user.FirstName,
+            SecondName = user.SecondName,
+            IsAdmin = user.IsAdmin
+        };
+        
+        return response;
+    }
+
+    public async Task<string> GetUserEmailAsync(string token)
+    {
+        using (var connection = new MySqlConnection(_connectionString))
+        {
+            connection.Open();
+            var query = "SELECT email FROM token WHERE refreshToken = @token";
+            var result = await connection.QueryAsync<string>(query, new { token });
+            return result.FirstOrDefault();
+        }  
+    }
+
+    public bool ValidateToken(string token)
+    {
+        var tokenHandler = new JwtSecurityTokenHandler();
+        var key = Encoding.ASCII.GetBytes(_key);
+
+        try
+        {
+            tokenHandler.ValidateToken(token, new TokenValidationParameters
+            {
+                ValidateIssuerSigningKey = true,
+                IssuerSigningKey = new SymmetricSecurityKey(key),
+                ValidateIssuer = false,
+                ValidateAudience = false,
+                ClockSkew = TimeSpan.Zero
+            }, out SecurityToken validatedToken);
+
+            return true;
+        }
+        catch
+        {
+            return false;
+        }
+    }
+    
+    public bool DeleteRefreshToken(string token)
+    {
+        try
+        {
+            using (var connection = new MySqlConnection(_connectionString))
+            {
+                connection.Open();
+                var sql = "DELETE FROM token WHERE refreshToken = @refreshToken";
+                connection.Execute(sql, new { refreshToken=token });
+            }   
+            
+            return true;
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine($"An error occurred: {ex.Message}");
+            return false;
+        }
+    }
+
+    private async void _WriteTokenInDb(string token, string email)
+    {
+        using (var connection = new MySqlConnection(_connectionString))
+        {
+            connection.Open();
+            var parameters = new { Email = email, RefreshToken = token };
+            var sql = @"
+                        INSERT INTO token (refreshToken, email)
+                        VALUES (@RefreshToken, @Email)
```

Comment:
I would store userId not email here

Reply:
Depends on having email or ID as a user primary key

Reply:
When making requests or displaying data you can expose real email used by customer